plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/rule/sync/RefreshBasedSyncMechanism.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm.rule.sync;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.common.lifecycle.AbstractLifecycleComponent;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.plugin.wlm.WorkloadManagementPlugin;
+import org.opensearch.rule.InMemoryRuleProcessingService;
+import org.opensearch.rule.RuleEntityParser;
+import org.opensearch.rule.autotagging.FeatureType;
+import org.opensearch.rule.storage.AttributeValueStore;
+import org.opensearch.rule.storage.AttributeValueStoreFactory;
+import org.opensearch.threadpool.Scheduler;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.client.Client;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * This class pulls the latest rules from the RULES system index to update the in-memory view
+ */
+public class RefreshBasedSyncMechanism extends AbstractLifecycleComponent {
+    /**
+     * Setting name to control the refresh interval of synchronization service
+     */
+    public static final String RULE_SYNC_REFRESH_INTERVAL_SETTING_NAME = "wlm.rule.sync_refresh_interval";
+    /**
+     * Default value for refresh interval
+     */
+    public static final long RULE_SYNC_REFRESH_INTERVAL_DEFAULT = 5000;
+    /**
+     * Minimum value for refresh interval
+     */
+    public static final int MIN_SYNC_REFRESH_INTERVAL = 1000;
```

Comment:
For better readability, these should not be defined as ms, as we never planning to go below 1s